### ISSUE TITLE
Speedup and reduced memory usage for reinhard02

### ISSUE THREAD

**Issue by heckflosse** [LuminanceHDR/LuminanceHDR#84]
I created a new [branch](https://github.com/heckflosse/LuminanceHDR/tree/reinhard02speedup) in my fork with a speedup and reduced memory usage for reinhard02.

Memory usage is reduced by `width * height * 16 byte`

I used a 24 Mpixel file for my tests. Here are the timings:
```
                                  before                    after
use scales disabled               4450 ms                   200 ms
use scales enabled               53500 ms                  9200 ms
```

For `use scales enabled` I used fftw wisdom, means the first run for a particular image size (width, height) will be slower (not slower than before) than the subsequent runs.
Currently the wisdom is saved to a file with hardcoded filename `lhdrwisdom.fftw`.

**Comment by Benitoite** [comment]
![screen shot 2017-11-30 at 10 51 04 am](https://user-images.githubusercontent.com/17919239/33449058-85774cb8-d5bc-11e7-8893-87375d97ae16.jpg)
Your four recent speedup branches merged on my fork and built for mac, all four working great. https://filebin.net/uarumuwfxhbv3jxu/Luminance_HDR_2.5.2-SNAPSHOT-Test1d.dmg

**Comment by heckflosse** [comment]
@Benitoite Thanks a lot for providing that build.
Tomorrow I will create a new branch with a speedup for Pattaniak.
Here first results (measured with my 24 Mpixel test image):
```
                            before          after
pattanaik non local        60600 ms        1480 ms
pattanaik local           300000 ms       11390 ms
```

**Comment by heckflosse** [comment]
Fixed with 0b394f7
